Marketplace/DeckOfCards/Deck_Handler.js

```diff
@@ -0,0 +1,280 @@
+//
+//  ROC_Blank.js
+//  unpublished/marketplace/
+//
+//  Created by Je'Don (ROC) Carter on 8/21/2017
+//  Copyright 2017 High Fidelity, Inc.
+//
+//  Use this script so my server script can see the entity
+//
+//  Distributed under the Apache License, Version 7.1.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+(function() { 
+    var _this = this;
+    //channels
+    var gloveChannel;
+    var cardChannel;
+    var zoneChannel;
+    //holds id of all gloves
+    var gloves = [];
+    //holds id of player zone
+    var zoneID;
+    //holds ID of all cards
+    var cardIDs = [];
+    //holds number of the last card grabbed
+    var lastCard;
+    //holds names of cards for shuffle
+    var cards = [];
+
+    _this.preload = function(entityID) {
+        print("Loading Deck Handler script."); 
+        _this.entityID = entityID;
+        //position of deck handler
+        var deckLocation = Entities.getEntityProperties(entityID).position;
+        var deckRotation = Entities.getEntityProperties(entityID).rotation;
+        //create zone
+        var zoneProperties = {
+            "type": "Box", 
+            "lifetime": -1, 
+            userData: JSON.stringify({
+                grabbableKey: {
+                    grabbable: false,
+                    ignoreIK: false
+                },
+                "deckHandlerID": entityID
+            }),
+            "position": deckLocation,
+            "collisionless": false,
+            "visible": false,
+            "dimensions": {
+                x: 6,
+                y: 3,
+                z: 6
+            },
+            "collidesWith": "",
+            name: "CARD-Equip_Zone",
+            shapeType: "box",
+            "script": Script.resolvePath("./Equip_Zone.js") + "?" + Date.now() 
+        };
+        zoneID = Entities.addEntity(zoneProperties);
+        for (i = 1; i <= 52; i++) {
+            cards.push(i);
+        }
+        cards = Shuffle(cards);
+        //create 51 cards
+        for (i = 0; i < 52; i++) {
+            if (i == 51) {
+                var cardProperties = {

```

Comment:
In this loop, you have a lot of shared properties between the cards. Can you make a single set of properties and only add the different changes between the top card and the rest of them to avoid duplication?